ls1intum/Artemis
# `Development`: Update line chart in course management detail view to ngx-charts

Fixes #3968



### Checklist
#### General


- [x] I tested **all** changes and their related features with **all** corresponding user types **locally**.
#### Client

### Motivation and Context


We has some problems with ng2-charts and chart.js in the past, so we decided to switch to ngx charts. 

### Description

As an example, I adapted the Course Detail View line chart to use the new library ngx charts.
The line changes are mostly the package-lock.json, there are only like 100-200 "active" line changes

**Before Testing:** Choose a course detail page for with you want to test the PR (make sure the line graph in this course detail page has some realistic values). Screenshot the Line graph to compare values after deploying the PR

### Steps for Testing

Prerequisites:
- 1 Instructor
- 2 Students

1. Before deploying, screenshot the line graph of an existing course detail page with realistic values
2. Deploy
3. Log in to Artemis as an Tutor or Instructor (or admin)
4. Navigate to Course Administration
5. Navigate into a course detail page of your choice
6. Look at the line chart and make sure now errors are thrown no abnormal behaviour is detected and the values math the values if your prior screenshot
7. Hover over the chart and verify that the tooltips given make sense and look good
8. Test the arrows on the right and left hand side of the graph and make sure data gets updated accordingly

### Review Progress





#### Code Review
#### Manual Tests

### Screenshots


Interface with new chart framework:
<img width="1422" alt="Screenshot 2021-10-07 at 14 27 02" src="https://user-images.githubusercontent.com/44401560/136384713-20a34934-f419-445d-a027-24363290a594.png">
